src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -72,6 +70,21 @@ def _post(self, request: Request):
         else:
             return Response("Unsupported version, we only support versions 1 to 3.", 400)
 
+    def _should_use_v3(self, request):
+        if request.GET.get("version") != "3":
+            return False
+        if not request.relay_request_data.get("fullConfig"):
+            # The v3 implementation can't handle partial configs. Relay by
+            # default request full configs and the amount of partial configs
+            # should be low, so we handle them per request instead of
+            # considering them v3.
+            return False
+        if request.relay_request_data.get("noCache"):
+            return False
```

Comment:
For these two it would be really nice to add a metric so we can see how many v3 requests of each type there are.  I imagine emitting these three:
```
metrics.incr('api.endpoints.relay.project_configs.request.v3', tags={'version': 'v3'})
metrics.incr('api.endpoints.relay.project_configs.request.v3', tags={'version': 'v2', 'reason': 'fullConfig'})
metrics.incr('api.endpoints.relay.project_configs.request.v3', tags={'version': 'v2', 'reason': 'noCache'})
```